src/renderer/components/external-player-settings/external-player-settings.js

```diff
@@ -42,6 +43,15 @@ export default Vue.extend({
     }
   },
   methods: {
+    handleUpdateExternalPlayer: async function (externalPlayer) {
+      const payload = {
+        isDev: this.isDev,
+        externalPlayer: externalPlayer
+      }
+      this.getExternalPlayerCmdArgumentsData(payload)
+      this.updateExternalPlayer(externalPlayer)
+    },
+
```

Comment:
I don't really like that you are reaching into the file every time the external player changes, but I'll tolerate it.

However, what I can't tolerate is that you re-set all the names and values when you change the external player **for no reason**. That should only be done when first setting them (on startup).

Make a separate function that only sets the correct cmd arguments for the chosen changed external player.

Reply:
Actually, the reason why I initially got the data on every change was, that I was under the assumption, that the file would be editable by users (in their installation).
Afterwards, I found out that that wasn't the case 🤷.
I changed it, so that it only gets the names, values, and cmd arguments when FreeTube is starting up.